app/api/routers/automatic.py

```
@@ -0,0 +1,127 @@
+                logger.info(f"Returned process to pool for session {session_id}")
+            else:
+                # Terminate direct process
+                try:
+                    if hasattr(proc, "poll") and proc.poll() is None:
+                        proc.terminate()
+                        proc.wait()
+                    elif hasattr(proc, "returncode") and proc.returncode is None:
+                        proc.terminate()
+                        await proc.wait()
+                    logger.info(f"Terminated direct process for session {session_id}")
+                except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)